app/javascript/mastodon/actions/statuses.js

```diff
@@ -84,25 +88,45 @@ export function redraft(status, raw_text) {
   };
 };
 
-export function deleteStatus(id, routerHistory, withRedraft = false) {
-  return (dispatch, getState) => {
-    let status = getState().getIn(['statuses', id]);
+export const editStatus = (id, routerHistory) => (dispatch, getState) => {
+  let status = getState().getIn(['statuses', id]);
 
-    if (status.get('poll')) {
-      status = status.set('poll', getState().getIn(['polls', status.get('poll')]));
-    }
+  if (status.get('poll')) {
+    status = status.set('poll', getState().getIn(['polls', status.get('poll')]));
+  }
+
+  dispatch(editStatusRequest());
+
+  api(getState).get(`/api/v1/statuses/${id}/source`).then(response => {
+    dispatch(editStatusSuccess());
+    ensureComposeIsVisible(getState, routerHistory);
+    dispatch(setComposeToStatus(status, response.data.text, response.data.spoiler_text));
+  }).catch(error => {
+    dispatch(editStatusFail(error));
+  });
+};
 
+export const editStatusRequest = () => ({
+  type: STATUS_EDIT_REQUEST,
+});
+
+export const editStatusSuccess = () => ({
+  type: STATUS_EDIT_SUCCESS,
+});
+
+export const editStatusFail = error => ({
+  type: STATUS_EDIT_FAIL,
+  error,
+});
```

Comment:
Those names may be slightly confusing as they are all to initiate status editing, while the names kind of suggest they are for submitting the edit.